File: 01-Front/Saniteau/client-src/model/facturation/Facturation.ts
import { Abonne } from "../abonne/Abonne";
import { FacturationLigne } from "./FacturationLigne";

export class Facturation {
    public idFacturation: number;
    public idCampagneFacturation: number;
    public abonne: Abonne;
    public dateFacturation: Date;
    public dateFacturationAsString: string;
    public facturationLignes: FacturationLigne[];
    public idDernierIndex: number;
    public payee: boolean;
}

File: 01-Front/Saniteau/Models/Facturation/Facturation.cs
using Saniteau.Models.Abonnés;
using System;
using System.Collections.Generic;
using System.Text;

namespace Saniteau.Models.Facturation
{
    public class Facturation
    {
        public int IdFacturation { get; set; }
        public int IdCampagneFacturation { get; set; }
        public Abonne Abonne { get; set; }
        public DateTime DateFacturation { get; set; }
        public List<FacturationLigne> FacturationLignes { get; set; }
        public int IdDernierIndex { get; set; }
        public bool Payee { get; set; }

        public Facturation() { }
        public Facturation(int idFacturation, int idCampagneFacturation, Abonne abonne, DateTime dateFacturation, List<FacturationLigne> facturationLignes, int idDernierIndex, bool payee) {
            IdFacturation = idFacturation;
            IdCampagneFacturation = idCampagneFacturation;
            Abonne = abonne;
            DateFacturation = dateFacturation;
            FacturationLignes = facturationLignes;
            IdDernierIndex = idDernierIndex;
            Payee = payee;
        }
    }
}
